airflow-core/src/airflow/models/connection.py

```diff
@@ -505,7 +504,9 @@ def get_connection_from_secrets(cls, conn_id: str) -> Connection:
 
         # If this is set it means are in some kind of execution context (Task, Dag Parse or Triggerer perhaps)
         # and should use the Task SDK API server path
-        if hasattr(sys.modules.get("airflow.sdk.execution_time.task_runner"), "SUPERVISOR_COMMS"):
+        from airflow.sdk.execution_time.task_runner import is_supervisor_comms_initialized
```

Comment:
this used hasattr on sys so we don't need to have airflow.sdk installed on the server components

Reply:
Or smth else - but this should be revised after we complete the isolation.